### PR TITLE
feat: integration setup

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/1292
Adding integration tests to the tx-wrapper will allow us to keep track of changing types
## How to test:

`yarn test`

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: src/__integrationtests__/transfer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+});
+
+describe("Checks the transfer method while connected to the chain", () => {
+  it("Creates a transfer", async () => {
+    const { block } = await rpcToLocalNode("chain_getBlock");
+
+    unsigned = methods.balances.transfer(
+      {
+        value: "90071992547409910",
+        dest: "4sejigvu6STHdYmmYf2SuN92aNp8TbrsnBBDUj7tMrJ9Z3cG",
```
Comment (LeonFLK) [inline src/__integrationtests__/transfer.spec.ts:57]: Maybe have this not hardcoded, but derive address from created account.

File: src/__integrationtests__/transfer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    });
+
+    tx = construct.signedTx(unsigned, signature, {
+      metadataRpc,
+      registry,
+    });
+
+    expectedTxHash = construct.txHash(tx);
+
+    actualTxHash = await rpcToLocalNode("author_submitExtrinsic", [tx]);
```
Comment (LeonFLK) [inline src/__integrationtests__/transfer.spec.ts:91]: Are we not comparing these two hashes like so:

```suggestion
    actualTxHash = await rpcToLocalNode("author_submitExtrinsic", [tx]);
    expect(expectedTxHash).toEqual(actualTxHash)
```

File: src/__integrationtests__/types.spec.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+});
+
+describe("checks the compatability of the current kilt types", () => {
+  it("checks the registry with the current types", () => {
+    getRegistry({
+      chainName: "mashnet-node",
+      specName: getRuntimeVersion.specName,
+      specVersion: getRuntimeVersion.specVersion,
+      metadataRpc,
+    });
```
Comment (LeonFLK) [inline src/__integrationtests__/types.spec.ts:31]: Again, do we want to write it like a test, or is checking, if this goes through, enough?

File: src/util.ts [hunk 4]
```diff
@@ -5,10 +5,10 @@
 /**
  * @ignore Don't show this file in documentation.
  */ /** */
-import { KeyringPair } from '@polkadot/keyring/types';
```
Comment (LeonFLK) [inline src/util.ts:8]: In the sdk we use single quotes, as per linting rules, why differing rules here?

**Review by LeonFLK — COMMENTED** [review]
We could change the way we test explicitly for success,
other wise LGTM.

File: src/util.ts (hunk 4 above)
Comment (Dudleyneedham) [reply src/util.ts:8]: Updating the rulings

File: src/__integrationtests__/types.spec.ts (hunk 3 above)
Comment (Dudleyneedham) [reply src/__integrationtests__/types.spec.ts]: We want to make sure it runs through. That being said having the tx hash as a final step makes sense.

File: src/__integrationtests__/transfer.spec.ts (hunk 1 above)
Comment (Dudleyneedham) [reply src/__integrationtests__/transfer.spec.ts]: Updated to have bob as the account

File: src/util.ts (hunk 4 above)
Comment (LeonFLK) [reply src/util.ts:8]: beautiful

**Review by LeonFLK — APPROVED** [review]
LGTM!!

**Pull request merged**